### ISSUE TITLE
Missing conversion action categories in the enums

### ISSUE THREAD

**Issue by alexandre67fr** [googleads/google-ads-php#329]
Hello,

Google Ads API [update was released yesterday](https://developers.google.com/google-ads/api/docs/release-notes#conversions). They introduced new conversion action categories, namely `PHONE_CALL_LEAD`, `ENGAGEMENT`, `IMPORTED_LEAD`, etc.
The values are missing from [`enums/conversion_action_category.proto`](https://github.com/googleapis/googleapis/blob/master/google/ads/googleads/v3/enums/conversion_action_category.proto), and therefore from `src/Google/Ads/GoogleAds/V3/Enums/ConversionActionCategoryEnum/ConversionActionCategory.php` , too.
The result is that I get an Exception when doing `search()` requests.

```
   Google\Protobuf\Internal\GPBDecodeException 

  Error occurred during parsing: Error occurred during parsing: Enum field only accepts integer or enum value name

  at vendor/google/protobuf/src/Google/Protobuf/Internal/Message.php:1309
    1305|         }
    1306|         try {
    1307|             $this->mergeFromJsonArray($array);
    1308|         } catch (\Exception $e) {
  > 1309|             throw new GPBDecodeException($e->getMessage());
    1310|         }
    1311|     }
    1312| 
    1313|     /**

  1   vendor/google/protobuf/src/Google/Protobuf/Internal/Message.php:779
      Google\Protobuf\Internal\Message::parseFromJsonStream(Object(Google\Protobuf\Internal\RawInputStream))

  2   vendor/google/gax/src/Transport/RestTransport.php:118
      Google\Protobuf\Internal\Message::mergeFromJsonString("{
  "results": [
    {
      "metrics": {
        "conversionLastReceivedRequestDateTime": "1970-01-01 01:00:00"
      },
      "conversionAction": {
        "resourceName": "customers/XXXXXXXX/conversionActions/XXXXXXXX",
        "id": "XXXXXXXX",
        "name": "Appels Extensions Annonces",
        "status": "ENABLED",
        "type": "AD_CALL",
        "category": "PHONE_CALL_LEAD",
        "ownerCustomer": "customers/XXXXXXXX",
        "includeInConversionsMetric": true,
        "clickThroughLookbackWindowDays": "60",
        "viewThroughLookbackWindowDays": "30",
        "valueSettings": {
          "defaultValue": 0,
          "defaultCurrencyCode": "EUR",
          "alwaysUseDefaultValue": true
        },
        "countingType": "MANY_PER_CLICK",
        "attributionModelSettings": {
          "attributionModel": "GOOGLE_ADS_LAST_CLICK",
          "dataDrivenModelStatus": "NEVER_GENERATED"
        },
        "phoneCallDurationSeconds": "15"
      }
    },

```

Thanks !

**Comment by PierrickVoulet** [comment]
Hi @alexandre67fr,

You are referring to the Google Ads API release, the client library for PHP is usually released a bit after. As stated in this Google Ads API blog [post](https://ads-developers.googleblog.com/2020/04/announcing-v31-of-google-ads-api-beta.html), it should be the done next week.

I am closing this issue but feel free to reopen if any further questions.

**Comment by PierrickVoulet** [comment]
@alexandre67fr I gave it a second thought and you are right, this is a bug. This Google Ads API release is considered as a "minor" release so it should not break client library users in any way.

**Comment by alexandre67fr** [comment]
Thanks, @PierrickVoulet 

Yes, I was a bit confused too. Version 3.0.0 and 3.1.0 of google-ads-php triggers the same exception, so I thought it has something to do with the API update.

In any case, this library is just awesome, thanks :)

**Comment by PierrickVoulet** [comment]
Thank you for the extra details. Most of the google-ads-php version numbers are aligned with the Google Ads API version but it is not always the case. This typically happens when we need to release some changes in google-ads-php even though the Google Ads API version remains the same.

In particular, the versions 3.0.0 and 3.1.0 of google-ads-php both are related to Google Ads API 3.0.0 so the confusion. The next release of google-ads-php should be 3.2.0 and it will support Google Ads 3.1.0.

Feel free to refer to the [CHANGELOG.md](https://github.com/googleads/google-ads-php/blob/master/CHANGELOG.md), this should help removing any confusion.

**Comment by PierrickVoulet** [comment]
@alexandre67fr do you happen to have request IDs that led to such failure? This will help me to quickly reproduce.

**Comment by alexandre67fr** [comment]
@PierrickVoulet Sure, the query was 

```mysql
            SELECT
                conversion_action.app_id,
                conversion_action.attribution_model_settings.attribution_model,
                conversion_action.attribution_model_settings.data_driven_model_status,
                conversion_action.category,
                conversion_action.click_through_lookback_window_days,
                conversion_action.counting_type,
                conversion_action.id,
                conversion_action.include_in_conversions_metric,
                conversion_action.name,
                conversion_action.owner_customer,
                conversion_action.phone_call_duration_seconds,
                conversion_action.resource_name,
                conversion_action.status,
                conversion_action.tag_snippets,
                conversion_action.type,
                conversion_action.value_settings.always_use_default_value,
                conversion_action.value_settings.default_currency_code,
                conversion_action.value_settings.default_value,
                conversion_action.view_through_lookback_window_days,
                metrics.conversion_last_conversion_date,
                metrics.conversion_last_received_request_date_time

            FROM
                conversion_action

```

with login customer ID 1249324803

**Comment by PierrickVoulet** [comment]
This issue seems specific to the REST transport (I could not reproduce using gRPC). REST is used as a fallback mechanism when the gRPC extension is not installed.

@alexandre67fr could you confirm that you do not have the gRPC extension installed? Would installing the gRPC extension (see [Requirements](https://github.com/googleads/google-ads-php#requirements) section) be a suitable workaround for you?

**Comment by alexandre67fr** [comment]
@PierrickVoulet yes, I can confirm that gRPC extension wasn't installed.

`Google\Protobuf\Internal\GPBDecodeException` doesn't occur anymore. Instead, I have 

```
  UnexpectedValueException 

  Enum Google\Ads\GoogleAds\V3\Enums\ConversionActionCategoryEnum\ConversionActionCategory has no name defined for value 11

  at vendor/googleads/google-ads-php/src/Google/Ads/GoogleAds/V3/Enums/ConversionActionCategoryEnum/ConversionActionCategory.php:79
    75| 
    76|     public static function name($value)
    77|     {
    78|         if (!isset(self::$valueToName[$value])) {
  > 79|             throw new UnexpectedValueException(sprintf(
    80|                     'Enum %s has no name defined for value %s', __CLASS__, $value));
    81|         }
    82|         return self::$valueToName[$value];
    83|     }

  1   app/PQP/Google/Ads/Import/ConversionActionImportable.php:166
      Google\Ads\GoogleAds\V3\Enums\ConversionActionCategoryEnum\ConversionActionCategory::name()
```

But that's no problem. Even though `ConversionActionCategory::name()` can't guess the correct name for value `11`, I can change the app to store the numeric value, so it will solve the issue.

It seems like Google Ads API is sending new values for `conversion_action_category.proto`, so this is not really a bug with the PHP library.

Thanks so much for your help.

**Comment by PierrickVoulet** [comment]
Quick update:

1. The version [3.2.0](https://github.com/googleads/google-ads-php/releases/tag/v3.2.0) of the client library is released and it contains the new enum values of `ConversionActionCategory`: you can upgrade to this version if you need to parse or use these enum values
1. The gax-php released a fix in version [1.3.1](https://github.com/googleapis/gax-php/releases/tag/1.3.1): when parsing an unknown enum value, REST ignores it and sets it to the default value of the enum instead of throwing an exception. In the case of `ConversionActionCategory`, the default value is [`UNSPECIFIED`](https://github.com/googleapis/googleapis/blob/master/google/ads/googleads/v3/enums/conversion_action_category.proto#L36). It is recommended to upgrade your project dependencies using the `composer update` command, this will avoid any further breaking changes due to new enum values in minor releases of the Google Ads API in the future.

This issue is now considered fixed, I am closing it.
